packages/react/modules/components/reconfigure-adapter/reconfigure-adapter.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import type { User, AdapterArgs } from '@flopflip/types';
+
+import React, { PureComponent, type Node } from 'react';
+
+import { withReconfiguration } from '../configure-adapter';
+
+type Props = {
+  exact?: boolean,
+  user: User,
+  reconfigure: (adapterArgs: AdapterArgs, { exact?: boolean }) => void,
+  children: React.Component<any>,
+};
+
+export class ReconfigureAdapter extends PureComponent<Props> {
+  static displayName = 'ReconfigureAdapter';
+
+  static defaultProps = {
+    exact: false,
+    children: null,
+  };
+
+  componentDidMount(): void {
+    return this.props.reconfigure(
+      {
+        user: this.props.user,
+      },
+      {
+        exact: this.props.exact,
+      }
+    );
+  }
+
+  componentDidUpdate(): void {
+    return this.props.reconfigure(
+      {
+        user: this.props.user,
+      },
+      {
+        exact: this.props.exact,
+      }
+    );
+  }
+
+  render(): Node {
+    return this.props.children
+      ? React.Children.only(this.props.children)
+      : null;
+  }
+}
+
+export default withReconfiguration()(ReconfigureAdapter);
```

Comment:
Hmm, wouldn't it make more sense to have the react component using the default implementation and the HOC reusing the react component?
Additionally, I would simply export the HOC from this module as a named export instead of a separate file.

Reply:
Thought about the latter too but liked working with the separate file. I kind of like these short files/components to not have to scroll around too much.

For the first: yeah, I wanted to always have a render-prop based version and the HoC using that. In this case the HoC only wraps the `Context.Consumer` to I didn't feel too much value in extracting that yet. Hope that makes sense.